public/app/features/alerting/unified/components/amroutes/AmRootRoute.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { FC, useState } from 'react';
+import { css } from '@emotion/css';
+import { GrafanaTheme, SelectableValue } from '@grafana/data';
+import { Button, useStyles } from '@grafana/ui';
+import { Receiver, Route } from 'app/plugins/datasource/alertmanager/types';
+import { AmRootRouteForm } from './AmRootRouteForm';
+import { AmRootRouteRead } from './AmRootRouteRead';
+
+export interface AmRootRouteProps {
+  receivers: Array<SelectableValue<Receiver['name']>>;
+  route: Route | undefined;
```

Comment:
```suggestion
  route?: Route;
```

Reply:
I prefer the `route: Route | undefined` instead because the passed param might be undefined, not that the prop is optional 😄 .

Reply:
fair enough!